src/GraphQL.Tests/Bugs/Issue4060.cs

```diff
@@ -0,0 +1,228 @@
+                {
+                    // If the field exists in the dictionary, return its argument values.
+                    if (TryGetValue(key, out var value))
+                        return value!;
+                    throw new ArgumentOutOfRangeException(nameof(key));
+                }
+            }
+
+            // Property stubs required by `IReadOnlyDictionary`, not implemented in this example.
+            public IEnumerable<GraphQLField> Keys => throw new NotImplementedException();
+            public IEnumerable<IDictionary<string, ArgumentValue>> Values => throw new NotImplementedException();
+            public int Count => throw new NotImplementedException();
+
+            // Checks if the `FieldArgumentDictionary` contains a mapping for the specified field.
+            public bool ContainsKey(GraphQLField key) => _fieldMappings.ContainsKey(key);
+
+            // Returns an enumerator, though it is not implemented here.
+            public IEnumerator<KeyValuePair<GraphQLField, IDictionary<string, ArgumentValue>>> GetEnumerator() => throw new NotImplementedException();
+
+            // `TryGetValue` is the core logic for retrieving argument values for a given field.
+            public bool TryGetValue(GraphQLField key, out IDictionary<string, ArgumentValue> value)
+            {
+#pragma warning disable RCS1059 // Avoid locking on publicly accessible instance
+                // Locking is used to ensure thread safety when caching field-argument mappings.
+                lock (this)
+                {
+                    // If the last accessed field matches the current field, return cached arguments.
+                    if (key == _lastField)
+                    {
+                        value = _lastArgs!;
+                        return true;
+                    }
+                }
+
+                // Fetch the field's type information from the mapping.
+                var fieldType = _fieldMappings[key];
+
+                // Retrieve the arguments for the field using the `ExecutionHelper`.
+                // The `ExecutionHelper.GetArguments` method computes the arguments by using the field type and provided arguments in the request.
+                IDictionary<string, ArgumentValue>? value2 = ExecutionHelper.GetArguments(
+                    fieldType.Arguments,
+                    key.Arguments,
+                    _executionContext.Variables,
+                    _executionContext.Document,
+                    key,
+                    null
+                );
+
+                // If no arguments are found, return an empty immutable dictionary.
+                value = value2 ?? ImmutableDictionary<string, ArgumentValue>.Empty;
+
+                // Cache the current field and its arguments to avoid recomputation.
+                lock (this)
+                {
+                    _lastField = key;
+                    _lastArgs = value;
+                }
+
+                return true;
```

Comment:
The method never returns `false`... Should the method signature change?
```c#
public bool TryGetValue(GraphQLField key, out IDictionary<string, ArgumentValue> value)
```
To
```c#
public IDictionary<string, ArgumentValue> GetValue(GraphQLField key)
```

Reply:
no, this method signature comes from `IReadOnlyDictionary...`